assets/js/modules/analytics/datastore/accounts.js

```diff
@@ -130,19 +136,23 @@ export const reducer = ( state, { type, payload } ) => {
 };
 
 export const resolvers = {
-	*getAccounts() {
+	*getAccounts( { force = false } = {} ) {
```

Comment:
@aaemnnosttv What's the reason for introducing this parameter? Why can't we invalidate the resolver instead?

Reply:
This is something Matthew and I worked out. It still requires that the resolver is invalidated to fetch again, but basically most of the resolvers return early now if data is already set in the store. This would prevent `getAccounts` from running again when we want to do the request anyways. For that reason, we need to allow for this kind of optional `force` parameter.

Part of the reason for this is testing - mocking fetch requests for the resolvers is a lot more work than dispatching `receive*` with the data that should be there to make that unnecessary for most cases that don't test fetch specifically. It's also important for storybook where we can't use jest for mocking.

Reply:
I suppose one alternative might be allowing for a falsy value to be set by `receiveAccounts` that would allow for the condition to pass here without requiring `force`. Or, we could have a dedicated action for `resetAccounts` or something which would set `accounts` back to `undefined`, and invalidate the resolver.

Reply:
I'm honestly starting to question whether having the resolvers bail early if there's data already is the right call. It kind of defeats one of the purposes of why resolvers overall exist.

As far as I recall, we mainly decided to do that to prevent property and profile requests to fire when e.g. `fetchAccountsPropertiesProfiles` had already populated them - is that correct? I think we should consider a different path.

Checking whether the values already exist in the resolver seems a bit off for the above reasons, and the need for a `force` parameter confirms that. We should be able to "force" things by invalidating resolvers.

Let's discuss this further.